src/transport/IonGasTransport.cpp

```diff
@@ -0,0 +1,388 @@
+//! @file IonGasTransport.cpp
+
+// This file is part of Cantera. See License.txt in the top-level directory or
+// at http://www.cantera.org/license.txt for license and copyright information.
+
+#include "cantera/transport/IonGasTransport.h"
+#include "cantera/numerics/polyfit.h"
+#include "cantera/base/stringUtils.h"
+#include "MMCollisionInt.h"
+
+namespace Cantera
+{
+IonGasTransport::IonGasTransport() :
+    m_kElectron(npos)
+{
+}
+
+void IonGasTransport::init(thermo_t* thermo, int mode, int log_level)
+{
+    m_thermo = thermo;
+    m_nsp = m_thermo->nSpecies();
+    m_mode = mode;
+    m_log_level = log_level;
+    // make a local copy of species charge
+    for (size_t k = 0; k < m_nsp; k++) {
+        m_speciesCharge.push_back(m_thermo->charge(k));
+    }
+
+    // Find the index of electron
+    if (m_thermo->speciesIndex("E") != npos ) {
+        m_kElectron = m_thermo->speciesIndex("E");
+    }
+
+    // Find indices for charge of species
+    for (size_t k = 0; k < m_nsp; k++){
+        if (m_speciesCharge[k] != 0){
+            if (k != m_kElectron) {
+                m_kIon.push_back(k);
+            }
+        } else {
+            m_kNeutral.push_back(k);
+        }
+    }
+    // set up Monchick and Mason parameters
+    setupMM();
+    // set up n64 parameters
+    setupN64();
+    // setup  collision integrals
+    setupCollisionIntegral();
+    m_molefracs.resize(m_nsp);
+    m_spwork.resize(m_nsp);
+    m_visc.resize(m_nsp);
+    m_sqvisc.resize(m_nsp);
+    m_phi.resize(m_nsp, m_nsp, 0.0);
+    m_bdiff.resize(m_nsp, m_nsp);
+    m_cond.resize(m_nsp);
+
+    // make a local copy of the molecular weights
+    m_mw = m_thermo->molecularWeights();
+
+    m_wratjk.resize(m_nsp, m_nsp, 0.0);
+    m_wratkj1.resize(m_nsp, m_nsp, 0.0);
+    for (size_t j = 0; j < m_nsp; j++) {
+        for (size_t k = j; k < m_nsp; k++) {
+            m_wratjk(j,k) = sqrt(m_mw[j]/m_mw[k]);
+            m_wratjk(k,j) = sqrt(m_wratjk(j,k));
+            m_wratkj1(j,k) = sqrt(1.0 + m_mw[k]/m_mw[j]);
+        }
+    }
+
+    // set flags all false
+    m_visc_ok = false;
+    m_viscwt_ok = false;
+    m_spvisc_ok = false;
+    m_bindiff_ok = false;
+    m_spcond_ok = false;
+    m_condmix_ok = false;
+}
```

Comment:
I'm not quite sure how to achieve this, but it would be nice if we didn't have to duplicate so much code from `GasTransport::init` here.

Reply:
It seems like only update_T use init, so maybe we can just delete those.